FreeSurfer/FreeSurferPipelineNHP.sh

```diff
@@ -0,0 +1,1219 @@
+
+# ----------------------------------------------------------------------
+log_Msg "Retrieve positional parameters"
+# ----------------------------------------------------------------------
+SubjectDIR="${1}"
+SubjectID="${2}"
+T1wImage="${3}"       # Irrelevant if '--existing-subject' flag is set
+T1wImageBrain="${4}"  # Irrelevant if '--existing-subject' flag is set
+T2wImage="${5}"       # Irrelevant if '--existing-subject' flag is set
+recon_all_seed="${6}"
+
+#RunMode="$p_runmode"
+#SPECIES="$p_species"
+RunMode="${RunMode:-1}"
+SPECIES="${Species}"
+
+# Species-specific environments imported from SPECIES4FS.cnf
+export SPECIES
+source ${HCPPIPEDIR}/FreeSurfer/custom/SetUpFSNHP.sh
+ScaleSuffix="_scaled"
+
+# Convert the --runmode string argument into a numeric code
+case "$RunMode" in
+	Default)
+		RunMode=1
+		;;
+	FSinit)
+		RunMode=1
+		;;
+	FSbrainseg)
+		RunMode=2
+		;;
+	FSsurfinit)
+		RunMode=3
+		;;
+	FShires)
+		RunMode=4
+		;;
+	FSfinish)
+		RunMode=5
+		;;
+	*)
+		echo "Error: invalid runmode '$RunMode'"
+		exit 1
+		;;
+esac
+
+## MPH: Hack!
+# For backwards compatibility, continue to allow positional specification of parameters for the above set of 6 parameters.
+# But any new parameters/options in the script will only be accessible via a named parameter/flag.
+# Here, we retrieve those from the global variable that was set in get_options()
+if [ "${flair}" = "TRUE" ] ; then
+	flair=${flair}
+fi
+if [ "${p_existing_subject}" = "TRUE" ] ; then
+	existing_subject=${p_existing_subject}
+fi
+if [ ! -z "${p_extra_reconall_args}" ] ; then
+	extra_reconall_args="${p_extra_reconall_args}"
+fi
+if [ ! -z "${p_conf2hires}" ] ; then
+	conf2hires=${p_conf2hires}
+fi
+
+# ----------------------------------------------------------------------
+# Log values retrieved from positional parameters
+# ----------------------------------------------------------------------
+log_Msg "SubjectDIR: ${SubjectDIR}"
+log_Msg "SubjectID: ${SubjectID}"
+log_Msg "T1wImage: ${T1wImage}"
+log_Msg "T1wImageBrain: ${T1wImageBrain}"
+log_Msg "T2wImage: ${T2wImage}"
+log_Msg "recon_all_seed: ${recon_all_seed}"
+log_Msg "flair: ${flair}"
+log_Msg "existing_subject: ${existing_subject}"
+log_Msg "extra_reconall_args: ${extra_reconall_args}"
+log_Msg "conf2hires: ${conf2hires}"
+log_Msg "runmode: ${RunMode}"
+log_Msg "SPECIES: ${SPECIES}"
+# ----------------------------------------------------------------------
+log_Msg "Figure out the number of cores to use."
+# ----------------------------------------------------------------------
+# Both the SGE and PBS cluster schedulers use the environment variable NSLOTS to indicate the
+# number of cores a job will use. If this environment variable is set, we will use it to
+num_cores=0
+# determine the number of cores to tell recon-all to use.
+
+if [[ -z ${NSLOTS} ]] ; then
+	num_cores=8
+else
+	num_cores="${NSLOTS}"
+fi
+log_Msg "num_cores: ${num_cores}"
+
+if [ "${existing_subject}" != "TRUE" ] ; then
```

Comment:
With the current `opts_StringToBool` sanitizing and the longitudinal naming, this should be:
```suggestion
if ((! existing_session)) ; then
```

Reply:
I fixed it. 5bd66b17068175d90975a30eb7d1fe0aa204ffc8